tests/security/grub_auth/grub_authorization.pm

```diff
@@ -0,0 +1,74 @@
+# Copyright (C) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: grub2 supports restricting access to boot menu entries when building their
+#          images/appliances,so that only specified users can boot selected menu entries.
+#
+# Maintainer: rfan1 <richard.fan@suse.com>
+# Tags: poo#81721, tc#1768659
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use base 'consoletest';
+
+my $sup_user     = 'admin';
+my $sup_passwd   = 'pw_admin';
+my $maint_user   = 'maintainer';
+my $maint_passwd = 'pw_maintainer';
+
+sub grub_auth_oper {
+    my $para = shift;
+    if ($para eq "operator") {
+        assert_screen("grub_auth_boot_menu_entry");
+        send_key("ret");
+        assert_screen("grub_auth_super_user_login");
+        type_string("$sup_user\n");
+        type_string("$sup_passwd\n");
+    }
+    elsif ($para eq "maintainer") {
+        assert_screen("grub_auth_boot_menu_entry_maintainer");
+        send_key("down");
+        send_key("ret");
+        assert_screen("grub_auth_maintain_user_login");
+        type_string("$maint_user\n");
+        type_string("$maint_passwd\n");
+    }
+}
+
+sub run {
+    select_console("root-console");
+
+    # Check disk name, partition number and fs_type for root file system
+    # Then create a new custom grub config file based on the users/passwords we definded
+    assert_script_run "wget --quiet " . data_url("grub_auth/create_custom_grub.sh");
+    assert_script_run("bash create_custom_grub.sh $sup_user $sup_passwd $maint_user $maint_passwd");
+
+    # Make sure both super user and maintainer can access the grub
+    foreach my $i ("grub_auth_oper", "maintainer") {
+        type_string("reboot\n");
+        grub_auth_oper($i);
+        assert_screen("linux-login", timeout => 90);
+        reset_consoles;
+        select_console("root-console");
+    }
```

Comment:
I think this will be better. Don't have duplicate code

Reply:
Thanks!